PUBLISH.md

```diff
@@ -1,33 +1,56 @@
-# Releasing a new version
+# Version management
 
-To be able to do `docker push` you need to have your Docker Hub account added to the dedis organisation.
-Here are the steps we use to make a conode binary release:
-```
-git checkout master
-git diff -> be sure you have no outstanding changes
-git log
-```
-review to be sure this is where the tag should be
+For the latest version of the conodes, please use 
+https://github.com/c4dt/byzcoin, which provides a continuous release of valid
+ conodes.
+
+For the `@dedis/kyber` and `@dedis/cothority` npm libraries, we provide a
+ development and a somewhat stable build.
+The development build is always up-to-date with the latest commit and can be
+ used with a reference to `@dedis/cothority@dev`:
+```bash
+npm i -s @dedis/cothority@dev
+```
+This pins the library to the latest available development version.
+
+## Releasing a new version for npm @dedis/*
+
+The npm-packages should follow semantic versioning.
+As we don't plan to have any big changes, the major-version is probably stuck
+ at `3`, except if we go the linux path and reach 3.9, a 4.0 could follow
+ , but more to avoid 3.10 than because anything else - or not.
+A minor-version increase should be done for a new stable functionality that
+ will be kept backward-compatible.
+A patch-version is mostly an irregular release whenever somebody thinks it's
+ important to have the latest code available.
+ 
+It is good to announce a release on the DEDIS/engineer slack channel.
+This allows others to know that a new release is about to happen and propose
+eventual changes.
+
+As the cothority depends on the kyber package, it's currently a bit akward to
+ update the package-version:
+1. update the kyber-version in `kyber/package.json` and push to master
+2. publish the new kyber-npm using `kyber/publish.sh`
+3. update the kyber-dependency in `cothority/package.json` and the cothority
+-version, which has to be the same as the kyber-version, push to master
+4. publish the new cothority-npm using `cothority/publish.sh`
+5. tag the latest commit using 
 ```
 git tag v3.4.6
```

Comment:
I would make it an "annotated" tag, possibly even signed.
From the `git tag` documentation:
```
Tag objects (created with -a, -s, or -u) are called "annotated" tags; they contain a creation date, the tagger name
and e-mail, a tagging message, and an optional GnuPG signature. Whereas a "lightweight" tag is simply a name
for an object (usually a commit object).

Annotated tags are meant for release while lightweight tags are meant for private or temporary object labels.
For this reason, some git commands for naming objects (like git describe) will ignore lightweight tags by default.
```